src/charactersheet/viewmodels/dm/encounter_sections/monster_section/index.js

```diff
@@ -1,309 +1,101 @@
-
-    self.populateMonster = function(label, value) {
-        var monster = DataRepository.monsters[label];
-
-        self.blankMonster().importValues(monster);
-        self.blankMonster().abilityScores().forEach(function(score, idx, _) {
-            score.name(monster.abilityScores[idx].name);
-            score.value(monster.abilityScores[idx].value);
-            score.shortName = ko.observable(score.name().substring(0,3).toUpperCase());
-        });
-        self.showDisclaimer(true);
-    };
-
-    /* Modal Methods */
-
-    self.modalFinishedOpening = function() {
-        self.showDisclaimer(false);
-        self.firstElementInModalHasFocus(true);
-    };
-
-    self.modalFinishedClosing = async function() {
-        self.openModal(false);
-        self.selectPreviewTab();
-
-        if (self.openEditModal()) {
-            const response = await self.currentEditItem().ps.save();
-            Utility.array.updateElement(self.monsters(), response.object, self.editItemIndex);
-        }
-
-        self.openEditModal(false);
-    };
-
-    self.selectPreviewTab = function() {
-        self.convertedDisplayUrl(Utility.string.createDirectDropboxLink(self.currentEditItem().sourceUrl()));
-        self.previewTabStatus('active');
-        self.editTabStatus('');
-    };
-
-    self.selectEditTab = function() {
-        self.editTabStatus('active');
-        self.previewTabStatus('');
-        self.editFirstModalElementHasFocus(true);
-    };
-
-    self.shouldShowExhibitButton = ko.pureComputed(function() {
-        return self._isConnectedToParty();
+    tagline = ko.pureComputed(() => {
+        const index = Fixtures.encounter.sections.mapsAndImages.index;
+        return this.encounter().sections()[index].tagline();
     });
 
-    self.pushModalFinishedClosing = function() {
-        self.selectedMonsterToPush(null);
-        self.openPushModal(false);
-    };
+    sorts() {
+        return {
+            ...super.sorts(),
+            'name asc': { field: 'name', direction: 'asc'},
+            'name desc': { field: 'name', direction: 'desc'},
+            'description asc': { field: 'description', direction: 'asc'},
+            'description desc': { field: 'description', direction: 'desc'},
+            'isExhibited asc': { field: 'isExhibited', direction: 'asc'},
+            'isExhibited desc': { field: 'isExhibited', direction: 'desc'},
+        };
+    }
 
-    self.pushModalToPlayerButtonWasPressed = function(monster) {
-        self.selectedMonsterToPush(monster);
-        self.openPushModal(true);
-    };
+    filteredAndSortedEntities = ko.pureComputed(() =>  (
+        SortService.sortAndFilter(this.entities(), this.sort(), null)
+    ), this);
 
-    /* Private Methods */
+    // Actions
 
-    self._dataHasChanged = async function() {
-        if (!self.encounterId()) {
-            return;
-        }
+    async toggleExhibit(monster) {
+        monster.isExhibited(!monster.isExhibited());
+        await monster.save();
```

Comment:
could there be an error returned from the save that should be checked before marking the monster as exhibited?